pyproject.toml

```diff
@@ -23,9 +23,9 @@ requires-python = ">=3.11"
 dependencies = [
   "invenio-celery>=2.0.0,<3.0.0",
   "invenio-db>=2.0.0,<3.0.0",
-  "invenio-jobs>=3.0.0,<4.0.0",
+  "invenio-jobs>=4.0.0,<5.0.0",
   "pysaml2>=7.5.0",
-  "validators>=0.34",
+  "validators>=0.21",
```

Comment:
why was it necessary to set down the limit?

Reply:
wait, but `>=0.21.0` is not equal to `^0.21.0`. the `^` version only works for `>=0.21.0,<0.22.0` or?

Reply:
yes

but most people (including us) use `>=...` everywhere, not `^...`
only that one dependency uses `^...` everywhere instead

now, it's using `^0.21` in particular, which - as you said - matches exactly the versions in `>=0.21.0,<0.22.0`,
so our restriction must allow at least one version in `>=0.21.0,<0.22.0` for locking to be resolvable

we *could* write for our restriction `^0.21` and that would work
but since we are using `>=...` (not `^...`), I went with `>=0.21.0` instead
this also allows locking to resolve to any `0.21.x` version